### PR TITLE
Negotiate Docker API version to support newest releases

### LINKED ISSUE #6989: Filebeat docker log enrichment& forwarding docker ee 17.06.2-ee-8
For confirmed bugs, please report:
- Version: 6.2.4
- Operating System: Windows Server 2016 Version 1607 (OS Build 14393.2068) 

- Steps to Reproduce:

1.  Install filebeat on docker host (windows server)
2. adjust beat configuration 
```yml
output.logstash:
  hosts: ["logstash:5044"]
filebeat.prospectors:
- type: log
  paths:
    - C:/ProgramData/docker/containers/*/*.log
  json.message_key: log
  json.keys_under_root: true
  processors:
  - add_docker_metadata:
      host: "tcp://172.30.128.1:2375"       
  document_type: log
  fields_under_root: true
  fields:
    environment: docker
    type: docker
```
3. Start filebeat service

**Expected result:**
Filebeat sends enriched docker logs to logstash configured.

**Actual result:**
2018-05-02T14:21:48.735+0200   ERROR  instance/beat.go:667      Exiting: Error in initing prospector: Error response from daemon: client version 1.22 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version

### PR DESCRIPTION
**Pull request by exekias**
:warning: WIP, do not merge

New Docker versions have deprecated our hardcoded client API version (1.22), this change updates the client to `docker.io/go-docker` and uses the new API version negotiation mechanism to ensure compatibility.

On the source repo change:

It seems there is a mess with tagging & vendoring docker after the change to docker-ce & moby (see https://github.com/moby/moby/issues/33989). I found https://github.com/docker/go-docker as a viable solution, although it seems to be WIP. I'm open to opinions and alternatives here.

Fixes #6989

### CODE REVIEW

**Comment by exekias** [comment]
LICENSE is not defined in https://github.com/docker/go-docker, will need to dig more for a solution

File: vendor/vendor.json [hunk 1]
```diff
@@ -2,6 +2,108 @@
...
+		{
+			"checksumSHA1": "fki/ItBkrrQ/+X7W6hH/2y6JhMg=",
+			"path": "docker.io/go-docker/api/types/registry",
+			"revision": "4daae26030ad00e348edddff9767924ae57a3b82",
+			"revisionTime": "2017-09-28T21:45:39Z"
+		},
+		{
+			"checksumSHA1": "N4QuNyYs3ToBqV6cR6lO6rvAJkE=",
+			"path": "docker.io/go-docker/api/types/strslice",
+			"revision": "4daae26030ad00e348edddff9767924ae57a3b82",
```
Comment (ruflin) [inline vendor/vendor.json]: I was initially curious if we should stick to a release here. Looking at https://github.com/docker/go-docker I looks like there are only 7 commits in this repo an all 8 month old. Is this really the lib that docker recommend moving forward?

**Comment by exekias** [comment]
Closed in favor of https://github.com/elastic/beats/pull/7165

File: vendor/vendor.json (hunk 1 above)
Comment (exekias) [reply vendor/vendor.json]: as I said, that lib is still work in progress, there is quite a mess with docker/moby and vendoring, I opted for keep using our current client, but implement negotiation with it, see #7165

**Pull request closed without merging**